### ISSUE TITLE
Placement Service not called for Actor Using Dotnet SDK and Docker-Compose

### ISSUE THREAD

**Issue by darthmolen** [dapr/dotnet-sdk#813]
When using the DotNetCore 6,  DotNet SDK, Docker Compose, custom bridge, the placement service never gets called.

docker-compose

```
version: '3.4'

services:
  rentalpropertyservice:
    image: ${DOCKER_REGISTRY-}rentalpropertyservice
    build:
      context: .
      dockerfile: K8S_Actor_API/Dockerfile
    ports: 
    - 50010:50001
    - "5124"
    depends_on:
      - dapr.statestores.redis
      - dapr.service.placement
    networks:
      - dapr

  
  rentalpropertyservice-dapr:
    image: "daprio/daprd:latest"
    command: [ "./daprd", "-app-id", "rentalpropertyservice", "-app-port", "5124", "-placement-host-address", "dapr.statestores.placement:50020", "-components-path", "/components"  ]
    volumes:
      - "./components/:/components"
    depends_on:
      - rentalpropertyservice
    network_mode: "service:rentalpropertyservice"

  dapr.service.placement:
    container_name: dapr.statestores.placement
    image: "daprio/dapr"
    command: ["./placement", "-port", "50020"]
    ports:
      - "50020:50020"
    networks:
      - dapr

  dapr.statestores.redis:
    container_name: dapr.statestores.redis
    image: "redis:alpine"
    ports:
      - "6380:6379"
    networks:
      - dapr
   
networks:
  dapr:
```

program.cs:
```
Log.Logger = new LoggerConfiguration()
    .WriteTo.Console()
    .CreateBootstrapLogger();

Log.Information("Starting up");


var builder = WebApplication.CreateBuilder(args);
builder.Host.UseSerilog((ctx, logger) => { logger.ReadFrom.Configuration(ctx.Configuration);  });


// Add services to the container.
// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
builder.Services.AddEndpointsApiExplorer();
builder.Services.AddSwaggerGen();
builder.Services.AddDaprClient(); ;
builder.Services.AddActors(options => {
    options.Actors.RegisterActor<RentalPropertyActor.Actors.RentalPropertyActor>();
});

var app = builder.Build();

app.UseSerilogRequestLogging();

// Configure the HTTP request pipeline.
if (app.Environment.IsDevelopment())
{
    app.UseSwagger();
    app.UseSwaggerUI();
}

app.MapActorsHandlers();


app.MapGet("/MarkPropertyReady/{OwnerId}/{PropertyId}/{UnitId}", async (string OwnerId, string PropertyId, string UnitId) => {

    if (string.IsNullOrWhiteSpace(OwnerId) || string.IsNullOrWhiteSpace(PropertyId))
        return Results.Problem("Owner Id or Property Id not filled out");

    var actorProxy = GetRentalPropertyActorAsync(OwnerId, PropertyId, UnitId);

    // Get UnitInfo, PropertyComplex from repo

    await actorProxy.PropertyReadyAsync(OwnerId, PropertyId, new RentalPropertyService.Model.UnitInfo(), new RentalPropertyService.Model.PropertyComplex());

    return Results.Ok();
})
    .WithName("MarkPropertyReady");
try
{
    Log.Information("Starting Webhost...");
    app.Run();

}
catch (Exception ex) 
{
    Log.Fatal(ex, "Unhandled Exception");
}
finally 
{
    Log.Information("Shut Down Complete");
    Log.CloseAndFlush();
}

IRentalPropertyActor GetRentalPropertyActorAsync(string OwnerId, string PropertyId, string UnitId)
{
    return ActorProxy.Create<IRentalPropertyActor>(RentalPropertyActor.Actors.RentalPropertyActor.GetActorId(OwnerId, PropertyId, UnitId), nameof(RentalPropertyActor));
}
```

when calling the actor endpoint via swagger, it says "Dapr.DaprApiException: actor runtime is not configured". Also, in the placement service, there is no log entries about a member update count. However, calling the /dapr/config on the service shows an entry, `{"entities":["RentalPropertyActor"],"reentrancy":{"enabled":false}}`

Code works when using the self-hosted mode.

`dapr run --app-id rentalpropertyservice --app-port 5124 --dapr-http-port 3500 dotnet run`

**Comment by simongullberg** [comment]
I've the exact same problem, did you figure it out?

**Comment by halspang** [comment]
@darthmolen - What logs do you see from Dapr? There are a couple reasons why the actor runtime may not be setup but the two main culprits would be:

- Dapr isn't able to hit your app when calling `dapr/config`
- The statestore isn't labeled as `actorstatestore: true`

**Comment by rheckart** [comment]
I'm getting the same issue. My statestore is labelled as `actorstatestore: true`

The error message on calling the actor method on the proxy (in my case, `await proxy.ProcessUpdate(update)`) is:

```
Dapr.DaprApiException: actor runtime is not configured
   at Dapr.Actors.DaprHttpInteractor.SendAsyncHandleUnsuccessfulResponse(Func`1 requestFunc, String relativeUri, CancellationToken cancellationToken)
   at Dapr.Actors.DaprHttpInteractor.SendAsync(Func`1 requestFunc, String relativeUri, CancellationToken cancellationToken)
   at Dapr.Actors.DaprHttpInteractor.InvokeActorMethodWithRemotingAsync(ActorMessageSerializersManager serializersManager, IActorRequestMessage remotingRequestRequestMessage, CancellationToken cancellationToken)
   at Dapr.Actors.Communication.Client.ActorRemotingClient.InvokeAsync(IActorRequestMessage remotingRequestMessage, CancellationToken cancellationToken)
   at Dapr.Actors.Client.ActorProxy.InvokeMethodAsync(Int32 interfaceId, Int32 methodId, String methodName, IActorRequestMessageBody requestMsgBodyValue, CancellationToken cancellationToken)
   at Dapr.Actors.Client.ActorProxy.ContinueWithResult[TRetval](Int32 interfaceId, Int32 methodId, Task`1 task)
   at Program.<>c__DisplayClass0_0.<<<Main>$>b__1>d.MoveNext() in [C:\Users\rheck\source\repos\TeleTodo\TeleTodo.MinimalWebService\Program.cs]():line 60
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Http.RequestDelegateFactory.ExecuteTaskResult[T](Task`1 task, HttpContext httpContext)
   at Microsoft.AspNetCore.Http.RequestDelegateFactory.<>c__DisplayClass46_3.<<HandleRequestBodyAndCompileRequestDelegate>b__2>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
   at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
   at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```
The logs when debugging looks like this:

```
> Executing task: daprd-debug <

> Executing command: daprd --app-id "teletodoapp" --app-port "3000" --components-path "./components" --dapr-http-port "3500" --metrics-port "9090" --placement-host-address "localhost:6050" <

time="2022-04-16T21:10:23.2404162-04:00" level=info msg="starting Dapr Runtime -- version 1.7.0 -- commit 3c4510a685470eb8a5403e9d9a451bc93e67e6c4" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.2404162-04:00" level=info msg="log level set to: info" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.2404162-04:00" level=info msg="metrics server started on :9090/" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.metrics type=log ver=1.7.0
time="2022-04-16T21:10:23.2440079-04:00" level=info msg="loading default configuration" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.2440079-04:00" level=info msg="standalone mode configured" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.2449702-04:00" level=info msg="app id: teletodoapp" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.2449702-04:00" level=info msg="mTLS is disabled. Skipping certificate request and tls validation" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.320053-04:00" level=info msg="local service entry announced: teletodoapp -> 192.168.50.248:63416" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.contrib type=log ver=1.7.0
time="2022-04-16T21:10:23.320053-04:00" level=info msg="Initialized name resolution to mdns" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3210694-04:00" level=info msg="loading components" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3338195-04:00" level=info msg="component loaded. name: pubsub, type: pubsub.redis/" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3348368-04:00" level=info msg="component loaded. name: telegramtodosecretstore, type: secretstores.local.file/v1" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3361337-04:00" level=info msg="waiting for all outstanding components to be processed" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.347034-04:00" level=info msg="detected actor state store: statestore" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3477499-04:00" level=info msg="component loaded. name: statestore, type: state.redis/" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3477499-04:00" level=info msg="all outstanding components processed" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3477499-04:00" level=info msg="gRPC proxy enabled" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3477499-04:00" level=info msg="enabled gRPC tracing middleware" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime.grpc.api type=log ver=1.7.0
time="2022-04-16T21:10:23.3477499-04:00" level=info msg="enabled gRPC metrics middleware" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime.grpc.api type=log ver=1.7.0
time="2022-04-16T21:10:23.3482597-04:00" level=info msg="API gRPC server is running on port 50001" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3482597-04:00" level=info msg="enabled metrics http middleware" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime.http type=log ver=1.7.0
time="2022-04-16T21:10:23.3482597-04:00" level=info msg="enabled tracing http middleware" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime.http type=log ver=1.7.0
time="2022-04-16T21:10:23.3487939-04:00" level=info msg="http server is running on port 3500" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3487939-04:00" level=info msg="The request body size parameter is: 4" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3487939-04:00" level=info msg="enabled gRPC tracing middleware" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime.grpc.internal type=log ver=1.7.0
time="2022-04-16T21:10:23.3493114-04:00" level=info msg="enabled gRPC metrics middleware" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime.grpc.internal type=log ver=1.7.0
time="2022-04-16T21:10:23.3493114-04:00" level=info msg="internal gRPC server is running on port 63416" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-16T21:10:23.3493114-04:00" level=info msg="application protocol: http. waiting on port 3000.  This will block until the app is listening on that port." app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
```

If Dapr isn't able to hit my app when calling `dapr/config`, what would that look like?

**Comment by halspang** [comment]
@rheckart - This line would be followed by more things:

`time="2022-04-16T21:10:23.3493114-04:00" level=info msg="application protocol: http. waiting on port 3000.  This will block until the app is listening on that port." app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0`

For example:

`INFO[0009] application discovered on port 5010           app_id=demo_actor instance=Hals-MacBook-Pro.local scope=dapr.runtime type=log ver=edge`

**Comment by rheckart** [comment]
I'm instantiating from a minimal API project and I don't see any entries past the `application protocol: http. waiting on port 3000.` log line. The code runs and debugs and does find the secrets manager.

```
using Dapr.Client;
using Dapr.Extensions.Configuration;
using Microsoft.AspNetCore.Mvc;
using TeleTodo.Base;
using TeleTodo.Service;
using Dapr.Actors;
using Dapr.Actors.Client;
using TeleTodo.Interfaces;
using Dapr.AspNetCore;

const string SECRET_STORE_NAME = "secretstore";
const string SECRET_TELEGRAM_URL_KEY = "telegramUrlToken";

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.
// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
builder.Services.AddEndpointsApiExplorer();
builder.Services.AddSwaggerGen();
builder.Services.AddDaprClient();

builder.Services.AddActors(options =>
{
    options.Actors.RegisterActor<ChatActor>();

    // Configure default settings
    options.ActorIdleTimeout = TimeSpan.FromMinutes(60);
    options.ActorScanInterval = TimeSpan.FromSeconds(30);
    options.DrainOngoingCallTimeout = TimeSpan.FromSeconds(60);
    options.DrainRebalancedActors = true;
    options.RemindersStoragePartitions = 7;
});

var app = builder.Build();
var daprClient = new DaprClientBuilder().Build();

// Configure the HTTP request pipeline.
if (app.Environment.IsDevelopment())
{
    app.UseSwagger();
    app.UseSwaggerUI();
}

// app.UseHttpsRedirection();
// app.MapControllers();
app.MapActorsHandlers();

var telegramUrlTokenSecret = await daprClient.GetSecretAsync(SECRET_STORE_NAME, SECRET_TELEGRAM_URL_KEY);

if (!telegramUrlTokenSecret.TryGetValue("telegramUrlToken", out string? telegramUrlToken)) throw new Exception($"{SECRET_TELEGRAM_URL_KEY} Secret not found");

app.MapPost("/{Token}", async ([FromBody] TelegramUpdate update, HttpRequest request) =>
{
    if (!request.Path.HasValue) return Results.Forbid();
    if (!request.Path.Value.Contains(telegramUrlToken)) return Results.Forbid();

    var actorId = new ActorId(update.Message.MessageId.ToString());
    var proxy = ActorProxy.Create<IChatActor>(actorId, "ChatActor");

    var response = await proxy.ProcessUpdate(update);

    return Results.Created($"{update.UpdateId}", response); // $"{update.UpdateId}", $"{update.UpdateId} is {telegramUrlToken}");
});

app.Run();
```

**Comment by halspang** [comment]
@rheckart - Where do you set your app port, I don't see anything that makes your app listen on port 3000 in that code sample. If you don't see anything past that log message it's most likely because dapr isn't able to contact your app.

The other thing that could be happening here is your app could be crashing because you try and get a secret from Dapr before it's initialized. Your app failing to come up could cause dapr to not finish as well. In this case, there's a method in the DaprClient for waiting for the sidecar and the configuration extension we vend also does this for you.

**Comment by rheckart** [comment]
@halspang I found out the the app port was set incorrectly in my launchsettings.json file. I changed the applicationUrl property in that file to use port 5001 and got farther, including the message about the actor runtime starting. So in my case, it looks like the Curious Case of Incorrect Port Assignment.

```
time="2022-04-19T18:34:41.6623525-04:00" level=info msg="application discovered on port 5001" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-19T18:34:41.6629852-04:00" level=warning msg="[DEPRECATION NOTICE] Adding a default content type to incoming service invocation requests is deprecated and will be removed in the future. See https://docs.dapr.io/operations/support/support-preview-features/ for more details. You can opt into the new behavior today by setting the configuration option `ServiceInvocation.NoDefaultContentType` to true." app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-19T18:34:41.9780762-04:00" level=info msg="application configuration loaded" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
time="2022-04-19T18:34:41.9791668-04:00" level=info msg="actor runtime started. actor idle timeout: 1h0m0s. actor scan interval: 30s" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime.actor type=log ver=1.7.0
time="2022-04-19T18:34:42.0117528-04:00" level=info msg="dapr initialized. Status: Running. Init Elapsed 4400.9219ms" app_id=teletodoapp instance=LAPTOP-EOLFRVG5 scope=dapr.runtime type=log ver=1.7.0
```
I have a different error occurring, but it may be me doing Dapr wrong:

`Dapr.DaprApiException: error invoke actor method: error finding address for actor type ChatActor with id 1365`

**Comment by halspang** [comment]
@rheckart - Glad we're making progress!

Do you see any messages like this: 

```
INFO[0004] placement tables updated, version: 76         app_id=generator instance=Hals-MacBook-Pro.local scope=dapr.runtime.actor.internal.placement type=log ver=edge
INFO[0006] placement tables updated, version: 77         app_id=generator instance=Hals-MacBook-Pro.local scope=dapr.runtime.actor.internal.placement type=log ver=edge
```

After your app starts? The versions are probably different, but you should see those if you're getting updates from the placement service.

If you do see that, what do you get from calling `/dapr/config` when pointing at your local app? That'll show you what actor type is registered.

**Comment by rheckart** [comment]
@halspang It was me doing Dapr wrong - I was messing with the placementHostAddress property for the VSCode tasks to see if that was the original issue. I had it set to the incorrect address and port number. Setting it to localhost:50005 fixed my issue - I'm all set at this point - thanks for the assistance! Hopefully it will help others.

**Comment by halspang** [comment]
@darthmolen - Were you able to get past your issue? If not, can you share some logs so we can troubleshoot it?

**Comment by halspang** [comment]
Closing this issue as it seems things were resolved. Feel free to re-open or create a new issue if you're still experiencing problems.
